pkg/codegen/internal/test/testdata/external-resource-schema/go/example/cat.go

```diff
@@ -230,15 +167,15 @@ func (o CatArrayOutput) ToCatArrayOutputWithContext(ctx context.Context) CatArra
 }
 
 func (o CatArrayOutput) Index(i pulumi.IntInput) CatOutput {
-	return pulumi.All(o, i).ApplyT(func(vs []interface{}) Cat {
-		return vs[0].([]Cat)[vs[1].(int)]
+	return pulumi.All(o, i).ApplyT(func(vs []interface{}) *Cat {
```

Comment:
I'll be curious if this here closes https://github.com/pulumi/pulumi/issues/7897

Reply:
I tried unit testing this but alas Go compilation is disabled because it generates reference against non-existing type:

```
        "requiredMetadataArray": {
          "type": "array",
          "items": {
            "$ref": "/kubernetes/v3.7.0/schema.json#/types/kubernetes:meta%2Fv1:ObjectMeta"
          }
        },
```

Becomes `RequiredMetadataArray metav1.ObjectMetaArrayInput` which is a dangling ref.